src/main/java/seedu/flashcard/ui/FlashcardStatsCard.java

```diff
@@ -0,0 +1,69 @@
+package seedu.flashcard.ui;
+
+import javafx.collections.FXCollections;
+import javafx.fxml.FXML;
+import javafx.scene.chart.PieChart;
+import javafx.scene.control.Label;
+import javafx.scene.layout.Region;
+import seedu.flashcard.model.flashcard.Flashcard;
+
+/**
+ * An UI component that displays information of a {@code Flashcard}.
+ */
+public class FlashcardStatsCard extends UiPart<Region> {
+
+    private static final String FXML = "FlashcardStatsCard.fxml";
+
+    /**
+     * Note: Certain keywords such as "location" and "resources" are reserved keywords in JavaFX.
+     * As a consequence, UI elements' variable names cannot be set to such keywords
+     * or an exception will be thrown by JavaFX during runtime.
+     *
+     */
+
+    public final Flashcard flashcard;
+
+    @FXML
+    private PieChart pieChart;
+
+    @FXML
+    private Label question;
+
+    @FXML
+    private Label reviewCount;
+
+    @FXML
+    private Label correctCount;
+
+    @FXML
+    private Label errorMessage;
+
+    /**
+     * Creates a {@code FlashcardListCard} with the given {@code Flashcard} and index to display.
+     */
+    public FlashcardStatsCard(Flashcard flashcard) {
+        super(FXML);
+        this.flashcard = flashcard;
+        question.setText(flashcard.getQuestion().toString());
+        int successFrequency = flashcard.getStatistics().getSuccessFrequency();
+        int reviewedFrequency = flashcard.getStatistics().getReviewFrequency();
+        int failureFrequency = reviewedFrequency - successFrequency;
+        double successRate = ((double) successFrequency / (double) reviewedFrequency) * 100;
+        double failureRate = 100 - successRate;
```

Comment:
Perhaps, you should abstract these calculations to methods

Reply:
i think they are only used once so there is no need for that.